src/core/InlineBlock.ts

```diff
@@ -8,85 +9,68 @@ import { isToken, type Token, TokenType } from './token';
-   * Begins inline block when lookahead through upcoming tokens determines
-   * that the block would be smaller than INLINE_MAX_LENGTH.
-   * @param  {Token[]} tokens Array of all tokens
-   * @param  {Number} index Current token position
-   */
-  beginIfPossible(tokens: Token[], index: number) {
-    if (this.level === 0 && this.isInlineBlock(tokens, index)) {
-      this.level = 1;
-    } else if (this.level > 0) {
-      this.level++;
-    } else {
-      this.level = 0;
-    }
-  }
-
-  /**
-   * Finishes current inline block.
-   * There might be several nested ones.
-   */
-  end() {
-    this.level--;
-  }
-
-  /**
-   * True when inside an inline block
-   */
-  isActive(): boolean {
-    return this.level > 0;
-  }
+  constructor(private expressionWidth: number) {}
 
   /**
    * Check if this should be an inline parentheses block
    * Examples are "NOW()", "COUNT(*)", "int(10)", key(`somecolumn`), DECIMAL(7,2)
    */
-  isInlineBlock(tokens: Token[], index: number): boolean {
-    let length = 0;
-    let level = 0;
+  public isInlineBlock(parenthesis: Parenthesis): boolean {
+    return this.inlineWidth(parenthesis) <= this.expressionWidth;
+  }
 
-    for (let i = index; i < tokens.length; i++) {
-      const token = tokens[i];
-      length += token.value.length;
+  private inlineWidth(parenthesis: Parenthesis): number {
+    let length = 2; // two parenthesis
 
-      if (this.isForbiddenToken(token)) {
-        return false;
+    for (const node of parenthesis.children) {
+      switch (node.type) {
+        case 'function_call':
+          length += node.nameToken.value.length + this.inlineWidth(node.parenthesis);
+          break;
+        case 'array_subscript':
+          length += node.arrayToken.value.length + this.inlineWidth(node.parenthesis);
+          break;
+        case 'parenthesis':
+          length += this.inlineWidth(node);
+          break;
+        case 'between_predicate':
+          length += this.betweenWidth(node);
+          break;
+        case 'clause':
+        case 'limit_clause':
+        case 'binary_clause':
+          return Infinity;
+        case 'all_columns_asterisk':
+          length += 1;
+          break;
+        case 'token':
+          length += node.token.value.length;
+          if (this.isForbiddenToken(node.token)) {
+            return Infinity;
+          }
+          break;
       }
 
       // Overran max length
       if (length > this.expressionWidth) {
-        return false;
-      }
-
-      if (token.type === TokenType.OPEN_PAREN) {
-        level++;
-      } else if (token.type === TokenType.CLOSE_PAREN) {
-        level--;
-        if (level === 0) {
-          return true;
-        }
+        return length;
       }
     }
-    return false;
+    return length;
+  }
+
+  private betweenWidth(node: BetweenPredicate): number {
+    return [node.betweenToken, node.expr1, node.andToken, node.expr2]
+      .map(token => token.value.length)
+      .reduce((a, b) => a + b);
```

Comment:
replace with `.join('').length`

Reply:
I'd rather avoid string-concatenation for simply computing the length.

Opted to instead extract a `sum()` function to eliminate the reduce.